### PR TITLE
CA-285349: replace -only-sr quicktest flag with -default-sr

### PR DESCRIPTION
**Pull request by minishrink**
As a result of the recent REQ-477 PR review, it was suggested to make the following switch:

- `-only-sr [name-label]`: former flag, only ran quicktests on SRs with a matching `name-label` field
- `-default-sr`: replacement flag, only runs quicktests on the pool's default SR

This flag is passed to all relevant quicktests which iterate through SRs. Currently this only includes the CBT and storage quicktests.

### CODE REVIEW

File: ocaml/quicktest/quicktest.ml [hunk 1]
```diff
@@ -857,7 +859,7 @@ let _ =
     "-iso-sr-path", Arg.String (fun x -> Quicktest_storage.iso_path := x), "Path to ISO SR";
     "-single", Arg.String (fun x -> tests_to_run := [ x ]), Printf.sprintf "Only run one test (possibilities are %s)" (String.concat ", " all_tests) ;
     "-all", Arg.Unit (fun () -> tests_to_run := all_tests), Printf.sprintf "Run all tests (%s)" (String.concat ", " all_tests);
-    "-only-sr", Arg.String (fun x -> Quicktest_storage.only_sr_name := Some x), "Run tests only on SR with specified name";
+    "-default-sr", Arg.Unit (fun () -> use_default_sr := true), "Only run SR tests on the pool's default SR";
```
Comment (gaborigloi) [inline ocaml/quicktest/quicktest.ml]: @edwintorok do we want to actually *remove* the `-only-sr` flag, or could we have both?

Comment (gaborigloi) [inline ocaml/quicktest/quicktest.ml]: In my opinion it's better if we do it the same way as before, and move the `use_default_sr` ref into `Quicktest_storage`. This way it won't need this parameter, and we won't have to pass it through all the tests that do SR selection.

File: ocaml/quicktest/quicktest.ml [hunk 2]
```diff
@@ -902,7 +904,7 @@ let _ =
           maybe_run_test "powercycle" (fun () -> with_vm s vm_powercycle_test);
           maybe_run_test "lifecycle" (fun () -> with_vm s Quicktest_lifecycle.test);
           maybe_run_test "copy" (fun () -> Quicktest_vdi_copy.start s sr);
-          maybe_run_test "import_raw_vdi" (fun () -> Quicktest_import_raw_vdi.start s);
```
Comment (edwintorok) [inline ocaml/quicktest/quicktest.ml]: Why this change?

File: ocaml/quicktest/quicktest_cbt.ml [hunk 3]
```diff
@@ -123,7 +123,7 @@ let vdi_data_destroy_test ~session_id ~vDI =
       ~msg:"VDI.data_destroy failed to update VDI.type";
     assert_cbt_status true ~session_id ~test ~vDI:snapshot
       ~msg:"VDI snapshot cbt_enabled field erroneously set to false";
-  (*  test_vdi_update ~session_id ~test snapshot; 
+  (*  test_vdi_update ~session_id ~test snapshot;
```
Comment (edwintorok) [inline ocaml/quicktest/quicktest_cbt.ml]: Did you intentionally reindent code here? If not might be better to send a "reindent XAPI" pull request to master fist by running `make reindent`, rather than reindenting files in each PR. Then we can rebase from master with `-Xignore-space-change` and continue our work without having to reindent things.
The checklist from the ticket contains a reindent item as well.

File: ocaml/quicktest/quicktest_storage.ml [hunk 4]
```diff
@@ -616,12 +616,11 @@ let foreach_sr session_id sr =
...
 
-let go s =
+let go s use_default_sr =
   let test = make_test "Listing available Storage Repositories" 0 in
   start test;
-  let srs = list_srs s in
+  let srs = list_srs s use_default_sr in
   debug test (Printf.sprintf "Found %d SRs" (List.length srs));
   success test;
-  if !only_sr_name = None then
```
Comment (edwintorok) [inline ocaml/quicktest/quicktest_storage.ml]: I think the reason for skipping the ISO test was that the ISO test was failing on our branch. If you tried this on your box and all tests pass, then deleting this line is correct.

Comment (minishrink) [reply ocaml/quicktest/quicktest_storage.ml]: Yes, it ran without problems on my dtbox

File: ocaml/quicktest/quicktest_cbt.ml (hunk 3 above)
Comment (minishrink) [reply ocaml/quicktest/quicktest_cbt.ml]: Not intentionally, I forgot to turn off my vim plugin which automatically strips whitespace.

File: ocaml/quicktest/quicktest.ml (hunk 1 above)
Comment (minishrink) [reply ocaml/quicktest/quicktest.ml]: I was under the impression we want to remove it.

Comment (edwintorok) [reply ocaml/quicktest/quicktest.ml]: -only-sr was added as a convenience flag while testing GFS2. Its just as convenient to change the default SR in Xencenter as passing the name, so I would not want to have two mechanisms if we do not need two.

**Comment by edwintorok** [comment]
Gabor is right, lets keep the global ref.

File: ocaml/quicktest/quicktest.ml (hunk 2 above)
Comment (minishrink) [reply ocaml/quicktest/quicktest.ml]: Ah, this happened because when I tested on my dtbox I commented out some of the tests so I could only run the relevant ones, I must've removed the semicolon while doing that, can undo it. Although I'm not sure why we include one for the final line within a try-with anyway, is it any different with or without?

Comment (edwintorok) [reply ocaml/quicktest/quicktest.ml]: If it already has a semicolon you get less changes when people add stuff to the end: you only need to add a new line, not modify the old to add the ; and add new line. As discussed in another PR the preferred style is to put the ; at the beginning of the line, but lets follow what the existing code already does here.

File: ocaml/quicktest/quicktest_cbt.ml (hunk 3 above)
Comment (minishrink) [reply ocaml/quicktest/quicktest_cbt.ml]: Reindentation PR (here[https://github.com/xapi-project/xen-api/pull/3506])

**Comment by minishrink** [comment]
I've removed any edits to quicktest_cbt.ml and kept all `default-sr` logic in quicktest_storage.ml, only changing the ref in quicktest.ml instead of passing it around to different quicktests.
